buildscripts/ci/tools/osuosl/publish.sh

```diff
@@ -115,14 +115,18 @@ fi
 
 # Delete old files
 if [ "$BUILD_MODE" == "nightly" ]; then
-    echo "Delete old MuseScoreNightly files"
-    number_to_keep=42 # includes the one we just uploaded and the symlink to it
+    echo "Delete old MuseScore-Studio-Nightly files"
+    number_of_days_to_keep=40
+    number_of_builds_per_day=2 # master and release branch
+    number_of_overhead_files_per_build=1 # `latest` symlink
     if [ "$OS" == "linux" ]; then
-        ((++number_to_keep)) # one extra for the zsync file
+        ((++number_of_overhead_files_per_build)) # zsync file
+        ((number_of_builds_per_day *= 3)) # x86_64, aarch64, armv7l
     elif [ "$OS" == "windows" ]; then
-        ((number_to_keep *= 2)) # two nightlies each night, namely portable and normal
+        ((number_of_builds_per_day *= 2)) # portable and normal
     fi
-    ssh -i $SSH_KEY musescore-nightlies@ftp-osl.osuosl.org "cd ~/ftp/$FTP_PATH; ls MuseScoreNightly* -t | tail -n +${number_to_keep} | xargs rm -f"
+    number_to_keep=$((number_of_days_to_keep * number_of_builds_per_day + number_of_overhead_files_per_build * number_of_builds_per_day))
+    ssh -i $SSH_KEY musescore-nightlies@ftp-osl.osuosl.org "cd ~/ftp/$FTP_PATH; ls {MuseScoreNightly,MuseScore-Studio-Nightly,MuseScore-4.}* -t | tail -n +$((number_to_keep+1)) | xargs rm -f"
```

Comment:
It's only the [Windows folder](https://ftp.osuosl.org/pub/musescore-nightlies/macos/4x/nightly/?C=M;O=D) that has `MuseScore-4.` files, so this pattern should also work, and be more restrictive in case the command runs in the wrong folder:

```Bash
cd ~/ftp/$FTP_PATH && ls -t MuseScore{-Studio-,}Nightly-* MuseScore-4.*-x86_64.{7z,paf.exe}
```

Other changes:
- Swapped `;` for `&&` in case `cd` fails (e.g. because the path doesn't exist).
- Moved `-t` immediately after `ls` to make it easier to see.

__Note:__ `ls` will return an error status when no `MuseScoreNightly*` or `MuseScore-4.*` files are found (same with your version of the command) but old `MuseScore-Studio-Nightly*` will still get deleted as intended, and the local script will continue processing since there's no `set -e`. If we want to, we can restrict the pattern to just `MuseScore-Studio-Nightly*` files tomorrow after the other files are deleted.

Reply:
Thanks, I applied both suggestions.